mlflow/metrics/base.py

```diff
@@ -14,3 +21,306 @@ class MetricValue:
     scores: List[float] = None
     justifications: List[float] = None
     aggregate_results: Dict[str, float] = None
+
+
+def _validate_text_predictions(predictions, metric_name):
```

Comment:
Can we move all of these metrics into a new file or else this file would be huge. It also be utils where we can import this.

In future, we can add other precanned metrics a similar manner as well